JSTests/microbenchmarks/math-log1p.js

```diff
@@ -0,0 +1,40 @@
+(function () {
+  var finiteResult = 0;
+  var posZeroResult = 0;
+  var negZeroResult = 0;
+  var infinityResult = 0;
+  var negInfinityResult = 0;
+  var nanResult = 0;
+  var values = [
+    0,
+    -0,
+    1,
+    -1 + Math.E,
+    -1,
+    -2,
+    Number.MAX_VALUE,
+    Number.MIN_VALUE,
+    Number.MAX_SAFE_INTEGER,
+    Number.MIN_SAFE_INTEGER,
+    Infinity,
+    -Infinity,
+    NaN
+  ];
+  for (var i = 0; i < testLoopCount; ++i) {
+    for (var j = 0; j < values.length; ++j) {
+      var output = Math.log1p(values[j]);
+      if (output !== output) nanResult++;
```

Comment:
Intentionally avoided using `Number.isNaN` to prevent potential side effects associated with its use